source/js/components/app.js

```diff
@@ -27,10 +27,10 @@ class App extends React.Component {
 
 	onAddInputClick() {
 		const {inputItems} = this.state;
-		const nextState = [...inputItems, {
+		const nextState = [{
```

Comment:
The naming of this variable tripped me up initially. I thought this contained the entirety of your new state, but really it's just the `inputItems` property of your new state.